src/containers/search_result.jsx

```diff
@@ -110,6 +115,32 @@ class SearchResult extends Component {
     });
   }
 
+  onClickSortColumn(e, highlight, key) {
+    const newSortBy = {
+      key: '',
+      order: '',
+    };
+    const { search: { query: { sortBy } } } = this.props;
+    if (highlight) {
+      if (sortBy.order === 'ascend') {
+        newSortBy.key = key;
+        newSortBy.order = 'descend';
+      } else {
+        newSortBy.key = key;
+        newSortBy.order = 'ascend';
+      }
+    } else {
+      newSortBy.key = key;
+      newSortBy.order = 'ascend';
+    }
+
+    this.props.actions.sortSearchResultsBy(newSortBy);
+    this.props.actions.searchArticles({
+      ...this.props.search.query,
+      newSortBy,
+    });
```

Comment:
Can the sortSearchResultsBy call the API individually? Then, you don't need to use `searchArticles` after calling `sortSearchResultsBy`.

Reply:
I think the action of sortSearchResultBy shouldn't call the search result API individually, because the this action should just do one thing which is change the sort method. The work of get search results should be left to the action of searchArticles.